config.php

```diff
@@ -14,6 +15,7 @@
         ['class' => TopMenu::className(), 'event' => TopMenu::EVENT_INIT, 'callback' => ['humhub\modules\mail\Events', 'onTopMenuInit']],
         ['class' => NotificationArea::className(), 'event' => NotificationArea::EVENT_INIT, 'callback' => ['humhub\modules\mail\Events', 'onNotificationAddonInit']],
         ['class' => ProfileHeaderControls::className(), 'event' => ProfileHeaderControls::EVENT_INIT, 'callback' => ['humhub\modules\mail\Events', 'onProfileHeaderControlsInit']],
+        ['class' => Application::class, 'event' => Application::EVENT_BEFORE_REQUEST, 'callback' => ['humhub\modules\mail\Events', 'onBeforeRequest']],
```

Comment:
Just as a thought: Could a separate new REST Module event be useful here?  I am not sure if we would have any advantages here. The event could e.g. collect the routes as Event attribute.

Reply:
Yes, my current solution doesn't require a new event trigger, and I find some modules already use the event `Application::EVENT_BEFORE_REQUEST` for other purposes. I need to investigate how to implement new separate event especially for the rest api url rules.

Reply:
We don't necessarily have to create a new event class, it was just an idea. Currently we very rarely use custom event classes, but it's probably cleaner.